goal_src/levels/common/plat.gc

```diff
@@ -5,3 +5,603 @@
+                                        )
+                                  (let
+                                   ((v0-0
+                                     (create-launch-control
+                                      (-> *part-group-id-table* 107)
+                                      obj
+                                      )
+                                     )
+                                    )
+                                   (set! (-> obj part) v0-0)
+                                   v0-0
+                                   )
+                                  )
+   )
+  )
+
+;; failed to figure out what this is:
+(defstate plat-startup (plat)
+  :virtual #t
+  :code
+  (behavior ()
+   (cond
+    ((logtest? (-> self path flags) (path-control-flag not-found))
+     (go-virtual plat-idle)
+     )
+    ((> (-> self sync period) 0)
+     (go-virtual plat-path-active)
+     )
+    (else
+     (go-virtual plat-idle)
+     )
+    )
+   (none)
+   )
+  )
+
+;; failed to figure out what this is:
+(defstate plat-idle (plat)
+  :virtual #t
+  :event
+  (the-as
+   (function process int symbol event-message-block object :behavior plat)
+   plat-event
+   )
+  :trans
+  (behavior ()
+   (dummy-20 self)
+   (none)
+   )
+  :code
+  (behavior ()
+   (plat-trans)
+   (rider-post)
+   (suspend)
+   (while #t
+    (when (not (-> self bouncing))
+     (plat-trans)
+     (rider-post)
+     (logior! (-> self mask) (process-mask sleep-code))
+     (suspend)
+     0
+     )
+    (while (-> self bouncing)
+     (plat-trans)
+     (rider-post)
+     (suspend)
+     )
+    )
+   (none)
+   )
+  )
+
+;; failed to figure out what this is:
+(defstate plat-path-active (plat)
+  :virtual #t
+  :event
+  (the-as
+   (function process int symbol event-message-block object :behavior plat)
+   plat-event
+   )
+  :exit
+  (behavior ()
+   (sound-stop (-> self sound-id))
+   (none)
+   )
+  :trans
+  (behavior ()
+   (set! (-> self path-pos) (if (logtest? (-> self fact options) 8)
+                             (get-current-phase (-> self sync))
+                             (get-current-phase-with-mirror (-> self sync))
+                             )
+    )
+   (eval-path-curve!
+    (-> self path)
+    (-> self basetrans)
+    (-> self path-pos)
+    'interp
+    )
+   (if
+    (<
+     (vector-vector-distance (-> self root-override trans) (ear-trans))
+     81920.0
+     )
+    (sound-play-by-name
+     (static-sound-name "eco-plat-hover")
+     (-> self sound-id)
+     1024
+     0
+     0
+     (the-as uint 1)
+     (-> self root-override trans)
+     )
+    )
+   (plat-trans)
+   (none)
+   )
+  :code
+  (the-as (function none :behavior plat) anim-loop)
+  :post
+  (the-as (function none :behavior plat) plat-post)
+  )
+
+;; definition for method 11 of type plat
+(defmethod copy-defaults! plat ((obj plat) (arg0 res-lump))
+  (set!
+   (-> obj mask)
+   (the-as process-mask (the-as int (logior #x80000 (-> obj mask))))
```

Comment:
Is this the output with the latest version of the decompiler? I think this should become a constant now.